services/device-registry-base/src/main/java/org/eclipse/hono/service/management/device/DeviceWithStatus.java

```diff
@@ -0,0 +1,82 @@
+/*******************************************************************************
+ * Copyright (c) 2019, 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.service.management.device;
+
+
+import org.eclipse.hono.util.RegistryManagementConstants;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * Device Information.
+ */
+@JsonIgnoreProperties()
+@JsonInclude(value = Include.NON_NULL)
+public class DeviceWithStatus extends Device {
```

Comment:
This class is the implementation of the JSON "API" in my eyes which is now twofold: on a GET the status property should be part of the response (which is why the device class still needs the "status" property) but on a PUT it shall be ignored (which is why it has `@JsonIgnoreProperties({ "status" })` on it). Because of this the intention actually was to use a different class, so that this is more clearly separated in our code: wherever the Device class is used, there is no possibility to update the status, wherever DeviceWithStatus is used the status may be updated.

We could use aggregation here having two properties in DeviceWithStatus: a device property (of type Device) and a status property (of type Status). Since the "status" JSON property needs to be on the same level as the other properties of a device, we would need to add `@JsonUnwrapped` on the device property.

Reply:
After doing some more research I found the `allowGetters` field in the `JsonIgnoreProperties` annotation. I think this is exactly what we need and actually could make the second class obsolete. 

I'll take a closer look if I can put this to work.

Reply:
Actually this worked and turned out to be the better solution in my opinion.